assets/dev/js/editor/controls/media.js

```diff
@@ -110,6 +113,35 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
 		}
 	},
 
+	/**
+	 * Hack to remove unwanted elements from modal.
+	 */
+	onFrameReady() {
+		const frameElement = document.querySelector( '.media-modal.wp-core-ui' );
+
+		const elementsToRemove = [
+			'#menu-item-insert',
+			'#menu-item-gallery',
+			'#menu-item-playlist',
+			'#menu-item-video-playlist',
+			'.embed-link-settings',
+		];
+
+		elementsToRemove.forEach( ( selector ) => {
+			const el = frameElement.querySelector( selector );
+
+			if ( el ) {
+				el.remove();
+			}
+		} );
+
+		// Remove elements from the URL upload tab.
+		frameElement.classList.add( 'unwanted-elements-removed' );
```

Comment:
```suggestion
		frameElement.classList.add( 'elementor-unwanted-elements-removed' );
```

Reply:
You're right. Changed it to a better name.